monitoring/src/deposit-monitor.ts

```diff
@@ -0,0 +1,52 @@
+import { Incident, Monitor as IncidentMonitor, Severity as IncidentSeverity } from "./incident"
+import type { DepositRevealedEvent } from "@keep-network/tbtc-v2.ts/dist/src/deposit"
+import type { Bridge } from "@keep-network/tbtc-v2.ts/dist/src/chain"
+import { BigNumber } from "ethers"
+
+const DepositRevealedIncident = (event: DepositRevealedEvent) => ({
+  title: "Deposit revealed",
+  severity: IncidentSeverity.Minor,
+  btcFundingTxHash: event.fundingTxHash.toString(),
+  btcFundingOutputIndex: event.fundingOutputIndex,
+  amount: event.amount.toNumber(),
+  ethRevealTxHash: event.transactionHash.toPrefixedString(),
+})
+
+const LargeDepositRevealedIncident = (event: DepositRevealedEvent) => ({
+  title: "Large deposit revealed",
+  severity: IncidentSeverity.Major,
+  btcFundingTxHash: event.fundingTxHash.toString(),
+  btcFundingOutputIndex: event.fundingOutputIndex,
+  amount: event.amount.toNumber(),
+  ethRevealTxHash: event.transactionHash.toPrefixedString(),
+})
+
+export class DepositMonitor implements IncidentMonitor {
+  private bridge: Bridge
+
+  constructor(bridge: Bridge) {
+    this.bridge = bridge
+  }
+
+  async check(fromBlock: number, toBlock: number): Promise<Incident[]> {
```

Comment:
I would consider separating *Notifications* from *Incidents* and have the severity only for *Incidents*.

I think an industry standard is to treat *Incidents* as something major. For example, in [Google metrics docs](https://cloud.google.com/monitoring/alerts/incidents-events) we have:

> An incident, also called an alert, is a record of the triggering of an alerting policy. Unless an alerting policy is snoozed or disabled, Cloud Monitoring opens an incident when a condition of an alerting policy is triggered. The incident contains information that you can use to investigate the cause of the notification.

Another example from [PagerDuty docs](https://support.pagerduty.com/docs/incidents):
> An incident represents a problem or an issue that needs to be addressed and resolved. Incidents trigger on services, and a service’s escalation policy prompts notifications to go out to on-call responders to remediate the issue.

In the context of tBTC v2, deposit revealed and large deposit revealed are notifications and they may have different importance. An incident would be, for example, optimistic minting requested too early.

Reply:
My reasoning: I wanted to have a single representation of a monitoring "event" and do the differentiation using severity (for now it's "minor", "major" and "critical" but I'm thinking about adding "notification" or "informational" as well). A single type is much easier to handle for monitors and receivers and helps keep the code simple. It looks like the name `Incident` is just unfortunate but I wanted to avoid naming it `Event` as it seems to be too broad and generic. I'll re-think the approach.

Reply:
Sure, I'll just call it `SystemEvent` as we also have chain events and the JS built-in `Event` type.

Reply:
See 053036454b1cd94f250698d6c438b439cd3ee593